### PR TITLE
fix #89,fix #90, fix #91 Refactored buildOptions, fixed bug with BOM and added support non-english characters in validator

### LINKED ISSUE #89: Enhancement of buildOptions function
I think parser will work a little bit faster if return defaultOptions when there are not options in buildOptions function, than set every option from defaultOptions in loop.

### LINKED ISSUE #90: Validating was crashed when XML data have byte order mark (BOM)
### Checklist

* [x] Are you running the latest version?
* [x] Have you included sample input, output, error, and expected output? 
* [x] Have you checked if you are using correct configuration?
* [x] Did you try online [tool](https://naturalintelligence.github.io/fast-xml-parser/)?

 
### Input
"\ufeff" + "<?xml version=\"1.0\" encoding=\"utf-8\" ?><КорневаяЗапись><Тэг>ЗначениеValue53456</Тэг></КорневаяЗапись>"

### Code
```js
const validator = require("../src/validator");

function validate () {
  const BOM = "\ufeff";
  let xmlData = BOM + "<?xml version=\"1.0\" encoding=\"utf-8\" ?><КорневаяЗапись><Тэг>ЗначениеValue53456</Тэг></КорневаяЗапись>".toString("utf8");
  let result = validator.validate(xmlData);
  console.log(result);
};
validate();
```

### Output
{ err: { code: 'InvalidChar', msg: 'char ﻿ is not expected .' } }

### expected data
{
            "КорневаяЗапись": {
                "Тэг":  "ЗначениеValue53456"
            }
}

**Would you like to work on this issue?**

- [x] Yes
- [ ] No

[Bookmark](https://github.com/NaturalIntelligence/fast-xml-parser/stargazers) this repository for further updates.

### LINKED ISSUE #91: Validator doesn't support cyrillic characters
### Checklist

* [x] Are you running the latest version?
* [x] Have you included sample input, output, error, and expected output? 
* [x] Have you checked if you are using correct configuration?
* [ ] Did you try online [tool](https://naturalintelligence.github.io/fast-xml-parser/)?

 
### Input
<КорневаяЗапись><Тэг>ЗначениеValue53456</Тэг></КорневаяЗапись>
### Code
```js
const validator = require('fast-xml-parser');
const options = {allowBooleanAttributes:  false, localeRange: "a-zA-Zа-яёА-ЯЁ"}
let xmlData = "<?xml version=\"1.0\" encoding=\"utf-8\" ?><КорневаяЗапись><Тэг>ЗначениеValue53456</Тэг></КорневаяЗапись>";
let result = validator.validate(xmlData, options);
console.log(result);

```

### Output
{ err: 
   { code: 'InvalidTag',
     msg: 'Tag КорневаяЗапись is an invalid name.' } }

### expected data
true

**Would you like to work on this issue?**

- [x] Yes
- [ ] No

[Bookmark](https://github.com/NaturalIntelligence/fast-xml-parser/stargazers) this repository for further updates.

### PR DESCRIPTION
**Pull request by serggonin**
# Purpose / Goal
Corrected returning value defaultOptions in buildOptions function instead of set every newOptions from defaultOptions in loop when there are not user's options. I think it give a little bit rise of perfomance (fix #89). Fixed bug in validation when XML data with byte order marker (BOM) (fix #90). Added support of non-english characters in XML validation (fix #91).

# Type
Enhancement, refactoring, bug fix

[x]Bug Fix
[x]Refactoring / Technology upgrade
[x]New Feature

### CODE REVIEW

File: src/validator.js [hunk 1]
```diff
@@ -18,6 +19,11 @@ exports.validate = function(xmlData, options) {
 
     const tags = [];
     let tagFound = false;
+    if (xmlData[0] === "\ufeff") {
```
Comment (amitguptagwl) [inline src/validator.js]: Please add a comment "check for byte order mark (BOM)"

**Comment by amitguptagwl** [comment]
Thanks for the PR. Will publish changes after generating the browser bundle soon.

**Comment by serggonin** [comment]
I hope you didn't forget about publish new version on npm?:relaxed:

**Comment by amitguptagwl** [comment]
I think I've forgot. Let me check :man_facepalming:

**Comment by amitguptagwl** [comment]
done

**Comment by serggonin** [comment]
Thank you!

**Pull request merged**